modules/apps/transfer/types/msgs.go

```diff
@@ -92,5 +99,48 @@ func (msg MsgTransfer) ValidateBasic() error {
 	if len(msg.Memo) > MaximumMemoLength {
 		return errorsmod.Wrapf(ErrInvalidMemo, "memo must not exceed %d bytes", MaximumMemoLength)
 	}
-	return ValidateIBCDenom(msg.Token.Denom)
+
+	for _, token := range msg.GetTokens() {
+		if !isValidToken(token) {
+			return errorsmod.Wrap(ibcerrors.ErrInvalidCoins, token.String())
```

Comment:
I find this confusing. We refer to `msg.Tokens` as a token, but it is of type coin. This is because the existing field is called `Token` and not coin. What do people think about:
* rename `msg.GetTokens()` to `msg.GetCoins` (lets only refer to it as a token if it is the token type)
* remove `isValidToken` and use `coin.Validate()` (I believe it does all the same checks)

current terminology:
* coin -> sdk.Coin
* token -> transfertypes.Token
* path -> "portid/channelid/portid/channelid/denom"
* trace -> {"portid/channelid", "portid/channelid"} 
* ibc denom -> ibc/{hash} (intended for internal chain usage, not used in packet data)

The one exception would be that `MsgTransfer.Tokens` would be []sdk.Coin

It'd be nice to be consistent with naming conventions. I've already noticed myself getting tripped up a few times because I read "token" but the actual type was a `sdk.Coin`

In addition, I think we could make `ValidateIBCDenom` private

Reply:
see my proposed change in https://github.com/cosmos/ibc-go/pull/6364